app/templates/components/template-filter.html

```diff
@@ -0,0 +1,125 @@
+{# 
+    This macro creates a filterable template list based on specified criteria. 
+    The macro requires three parameters:
+        - row_selector: A CSS selector for the rows to filter, i.e '.template-row'
+        - notification_types: list of notification types and text to display, i.e [{code: 'email', text: 'Email'}, {code: 'sms', text: 'SMS'}]
+        - template_categories: list of template categories and text to display, i.e [{code: 'status-update', text: 'Status Update'}, {code: 'password-reset', text: 'Password Reset'}]  
+        - notification_type_data_attribue: The data attribute to filter by notification type, i.e 'data-notification-type'  
+        - template_category_data_attribute: The data attribute to filter by template category, i.e 'data-template-category'
+    
+    NOTE: the data attributes need to be present on the rows indicated by `row-selector` for the filtering to work.  Example
+    of a complete row:
+    ```
+    <div class="template-row" data-notification-type="email" data-template-category="status-update">
+    ```
+#}
+{% macro template_filter(row_selector, notification_types, template_categories, notification_type_data_attribue, template_category_data_attribute) %}
+    {# TODO: move these to a CSS file #}
+    <style>
+        .template_filter .active {
+            background-color: #425A76;
+            color: white;
+        }
+        .template_filter details >div {
+            padding-left: 5px !important;
+            box-shadow: none !important;
+        }
```

Comment:
```suggestion
        .template_filter .active:not(:focus) {
            background-color: #425A76;
            color: white;
        }
        .template_filter details >div {
            padding-left: 5px !important;
            box-shadow: none !important;
        }
```

This doesn't seem ideal, but it could fix the issue with colour when an item is active and focused. Our css specificity for `a:link:focus` is not being super helpful here.

https://github.com/cds-snc/notification-admin/assets/5230720/579d2c97-836a-4479-811f-d8a66154ccce